github/github.go

```diff
@@ -2,121 +2,193 @@ package github
 
 import (
 	"encoding/json"
+	"errors"
 	"fmt"
-	"github.com/dgrijalva/jwt-go"
+	"net/url"
+	"strings"
+
 	"github.com/kintohub/utils-go/klog"
 	"github.com/valyala/fasthttp"
-	"strings"
-	"time"
 )
 
 type GithubInterface interface {
-	CallGithub(endpoint, verb, params, authToken string) ([]byte, error)
-	GenerateInstallationToken(installationId string) (string, error)
-	GenBearer(jwtToken string) string
+	GetListRepos(page int32, installationId, userAccessToken string) (*GithubRepositories, error)
+	GetUserInformation(userAccessToken string) (*GithubUserInfo, error)
+	CreateGithubAppToken(installationId string) (string, error)
+	GetUserAccessToken(code string) (string, error)
 }
 
 type github struct {
-	baseUrl       string
-	acceptHeader  string
-	appID         string
-	appPrivateKey []byte
+	appClientID     string
+	appClientSecret string
 }
 
-func New(baseUrl, acceptHeader, appID string, appPrivateKey []byte) GithubInterface {
+var (
+	BASE_API_URL             = "https://api.github.com"
+	BASE_URL                 = "https://github.com"
```

Comment:
comment - by doing that, u just make impossible the use of private github without redevelopment.

Reply:
yea i know, i think this is fine, supporting custom headers/base URLs was a big pain, especially found some endpoints accept different base URL and different accept headers